lib/tty/pager/basic.rb

```diff
@@ -43,50 +45,101 @@ def default_prompt
       #
       # @api public
       def page(text, &callback)
-        page_num = 1
-        leftover = []
-        lines_left = @height
+        write(text, &callback)
+        wait
+      end
+
+      def start
+        @running = true
+        @state = State.new(@height)
+
+        @state.queue = Queue.new
+        @state.thread = Thread.new do
+          loop do
+            message = @state.queue.pop
+            if message.nil?
+              sleep 0.1
+              next
+            end
+
+            type    = message[0]
+            payload = message[1]
+
+            case type
+            when :quit then break
+            when :print then output.print(payload)
+            when :prompt then instance_exec(payload, &@prompt)
+            else
+              raise "Unknown message type: #{type}"
+            end
+          end
+        end
+      end
+
+      def write(text, &callback)
+        start unless running?
 
         text.lines.each do |line|
           chunk = []
-          if !leftover.empty?
-            chunk = leftover
-            leftover = []
+          unless @state.leftover.empty?
+            chunk = @state.leftover
+            @state.leftover = []
           end
           wrapped_line = Strings.wrap(line, @width)
           wrapped_line.lines.each do |line_part|
-            if lines_left > 0
+            if @state.lines_left.positive?
               chunk << line_part
-              lines_left -= 1
+              @state.lines_left -= 1
             else
-              leftover << line_part
+              @state.leftover << line_part
             end
           end
-          output.print(chunk.join)
+          @state.queue << [:print, chunk.join]
 
-          if lines_left == 0
-            break unless continue_paging?(page_num)
-            lines_left = @height
-            if leftover.size > 0
-              lines_left -= leftover.size
+          if @state.lines_left.zero?
```

Comment:
I don't understand what that means and where I should use `next` to skip iteration. Is it the next line that uses `return`? Because I definitely do want to use `return` here, not `next`.

Reply:
Don't worry about Hound, probably need to configure hound/rubocop for this project.